ledger/ledger-api-common/src/test/suite/scala/com/digitalasset/ledger/api/tls/TlsConfigurationTest.scala

```diff
@@ -51,6 +56,56 @@ class TlsConfigurationTest extends AnyWordSpec with Matchers with BeforeAndAfter
 
       verifyOcsp(Enabled)
     }
+
+    "get an input stream from a plaintext private key" in {
+      // given
+      val keyFilePath = Files.createTempFile("private-key", ".txt")
+      Files.write(keyFilePath, "private-key-123".getBytes())
+      assume(Files.readAllBytes(keyFilePath) sameElements "private-key-123".getBytes)
+      val keyFile = keyFilePath.toFile
+      val tested = TlsConfiguration.Empty
+
+      // when
+      val actual: InputStream = tested.prepareKeyInputStream(keyFile)
+
+      // then
+      IOUtils.toString(actual, StandardCharsets.UTF_8) shouldBe "private-key-123"
+    }
+
+    "fail on missing secretsUrl when private key is encrypted ('.enc' file extension)" in {
+      // given
+      val keyFilePath = Files.createTempFile("private-key", ".enc")
+      Files.write(keyFilePath, "private-key-123".getBytes())
+      assume(Files.readAllBytes(keyFilePath) sameElements "private-key-123".getBytes)
+      val keyFile = keyFilePath.toFile
+      val tested = TlsConfiguration.Empty
+
+      // when
+      val e = intercept[IllegalStateException] {
+        val _: InputStream = tested.prepareKeyInputStream(keyFile)
+      }
+
+      // then
+      e.getMessage shouldBe "Unable to convert TlsConfiguration(true,None,None,None,None,REQUIRE,false,List()) to SSL Context: cannot decrypt keyFile without secretsUrl."
+    }
+
+    "attempt to decrypt private key using by fetching decryption params from an url" in {
+      // TODO PBATKO ideally I'd like to mock fetching and parsing the secrects but it seems hard to do.
+      // Alternative is to spin up a http server, but I'm already doing that in a different unit test
+      // given
+      val keyFilePath = Files.createTempFile("private-key", ".enc")
```

Comment:
@rautenrieth-da @nmarton-da I looked at mocking the fetch from the URL but couldn't do it as it is a method call on an object. Also, mocking plays nice when dependencies are injected but I don't think we are using any DI solution.

Do you think this test can stay as is; or have any hints how to approach mocking here?

Reply:
I modified the test to mock the URL. (Btw in order to mock`java.net.URL` class I added a mockito-extension file that enables mocking final classes)

Reply:
Wouldn't it be possible to find some URL that fails on all environments? For example localhost on a [reserved port number](https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers)? Something like `http://localhost:47/`?

Using a mock (or, according to [this article](https://www.martinfowler.com/bliki/TestDouble.html), a "stub") is fine too.

Reply:
I think I'll leave the current mock (stub) setup. I guess it's good enough.

Btw, I tried to search for on URL that is syntactically valid but one can never connect to it. The closest I found is `.invalid` domain (https://datatracker.ietf.org/doc/html/rfc6761). Not sure if I tried it correctly (~`http://localhost.invalid`) and eventually decided not to use it as it also changed the exception observed from ~connect exception to ~name resolution exception.